### PR TITLE
chore: initial setup of ESLint

### PR DESCRIPTION
**Pull request by kusigit**
### Description of change

Use ESLint to analyze source code and to improve code quality. 
[Prettier vs. Linters](https://prettier.io/docs/en/comparison.html)

This is just an idea. If you like it, some lines of code could be fixed automatically. The rest could be fixed manually step by step.

`✖ 4101 problems (0 errors, 4103 warnings)`
`0 errors and 606 warnings potentially fixable with the --fix option.`

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [x] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000` N/A
- [x] There are new or updated unit tests validating the change N/A
- [x] Documentation has been updated to reflect this change N/A
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -238,7 +240,7 @@
     "watch": "./node_modules/.bin/tsc -w",
     "package": "gulp package",
     "pack": "gulp pack",
-    "lint": "prettier --check \"./src/**/*.ts\" \"./test/**/*.ts\" \"./sample/**/*.ts\"",
+    "lint": "eslint . --ext .ts",
```
Comment (hinogi) [inline package.json:243]: good one there 👍 eslint for linting, prettier for formatting

Comment (hinogi) [reply package.json:243]: LGTM

A maintainer has to decide how package-lock updates should be handled.

**Review by pYassine — APPROVED** [review]

**Review by hinogi — APPROVED** [review]
LGTM

**Comment by pleerock** [comment]
Thank you for contribution! 🎉

**Pull request merged**